cmd/admin-handlers.go

```diff
@@ -244,32 +245,86 @@ func (a adminAPIHandlers) ServerInfoHandler(w http.ResponseWriter, r *http.Reque
 	if objectAPI == nil {
 		return
 	}
+	var jsonBytes []byte
+	var err error
+
+	vars := mux.Vars(r)
+	switch infoType := vars["infoType"]; infoType {
+
+	case "vault":
+		if GlobalKMS == nil {
+			writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrKMSNotConfigured), r.URL)
+			return
+		}
+		config, err := readServerConfig(ctx, objectAPI)
+		if err != nil {
+			writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+			return
+		}
+		endPoint := Environment.Get(EnvVaultEndpoint, config.KMS.Vault.Endpoint)
+		vaultName := Environment.Get(EnvVaultKeyName, config.KMS.Vault.Key.Name)
+		vaultType := Environment.Get(EnvVaultAuthType, config.KMS.Vault.Auth.Type)
+
+		var perm = madmin.Permission{}
+		kmsContext := crypto.Context{"MinIO admin API": "KMSKeyStatusHandler"} // Context for a test key operation
+		// Generate a new key using the KMS.
+		_, sealedKey, e := GlobalKMS.GenerateKey(vaultName, kmsContext)
+		if e == nil {
+			perm.Encryption = "OK"
+		} else {
+			perm.Encryption = "FAIL"
+		}
+
+		// Verify that we can indeed decrypt the (encrypted) key
+		_, e = GlobalKMS.UnsealKey(vaultName, sealedKey, kmsContext)
+		if e == nil {
+			perm.Decryption = "OK"
+		} else {
+			perm.Decryption = "FAIL"
+		}
+
+		vaultInfo := madmin.VaultInfo{
+			Endpoint: endPoint,
+			Name:     vaultName,
+			Type:     vaultType,
+			Perm:     perm,
+		}
```

Comment:
https://github.com/minio/minio/pull/7955/files introduces KMS key status - which reports error if encryption/decryption failed. You might want to make a note to refactor the vault switch case later to reuse code .  --json mode in mc should ideally show encryption/decryption error messages than a high level ok/fail.

Reply:
Added the error for encryption and decryption.